src/audio/pipeline.c

```diff
@@ -134,6 +134,11 @@ static int pipeline_for_each_comp(struct comp_dev *current,
 	list_for_item(clist, buffer_list) {
 		buffer = buffer_from_list(clist, struct comp_buffer, dir);
 
+		/* don't touch buffers from other pipelines */
```

Comment:
It's not entirely true. This is generic function used in all other pipeline traversal functions, so we shouldn't make this decision here. Such logic is handled in specific methods e.g. pipeline_prepare will execute across pipelines, pipeline_copy will execute only for the pipelines with the same scheduling id etc.

Reply:
@tlauda I've changed it, so that only pipeline_params and pipeline_prepare are limited to single pipeline, leaving the generic for_each unchanged.